play-media-native/screens/AddSport.tsx

```diff
@@ -0,0 +1,156 @@
+  const single = route?.params?.single;
+
+  const { contentItems, edit: editFields } = useContentItems();
+
+  const {
+    data: sports,
+    isLoading: isFetchingInitialSports,
+    refetch: refetchSports,
+    isRefetching: isRefetchingSports,
+  } = useSportsQuery();
+  const { calcScrollOffset } = useScrollOffset(true);
+  const initialSelectedIDs = useMemo(() => {
+    if (Array.isArray(contentItems[stateKey][fieldKey])) {
+      return contentItems[stateKey][fieldKey].map((item: Sport) => item.id);
+    }
+
+    return contentItems[stateKey][fieldKey]?.id || [];
+  }, [contentItems, fieldKey, stateKey]);
+  const [selectedSportIDs, setSelectedSportIDs] = useState<string[]>([initialSelectedIDs]);
+  const noneSelected = !selectedSportIDs?.length;
+
+  const edit = useCallback(
+    ({ key, value }: { key: string; value: Sport[] }) => {
+      editFields({ id: stateKey, key, value: [...contentItems[stateKey][key], ...value] });
+    },
+    [contentItems, editFields, stateKey]
+  );
+
+  const handleRefresh = useCallback(() => {
+    refetchSports();
+  }, [refetchSports]);
+
+  const onSelect = useCallback((sport: Sport) => {
+    setSelectedSportIDs((prevSelectedSportIDs) => {
+      if (prevSelectedSportIDs.includes(sport.id)) {
+        return prevSelectedSportIDs.filter((item) => item !== sport.id);
+      }
+
+      return [...prevSelectedSportIDs, sport.id];
+    });
+  }, []);
+
+  const onSelectSingle = useCallback(
+    (sport: Sport) => {
+      editFields({
+        id: stateKey,
+        key: route.params.key,
+        value: sport,
+      });
+
+      navigation.navigate(initialRoute);
+    },
+    [editFields, initialRoute, navigation, route.params.key, stateKey]
+  );
+
+  const onCancel = useCallback(() => {
+    navigation.goBack();
+  }, [navigation]);
+
+  const onSubmit = useCallback(() => {
+    if (!fieldKey || !initialRoute) {
+      return;
+    }
+
+    edit({
+      key: fieldKey,
+      value: sports.filter((item) => selectedSportIDs.includes(item.id)),
+    });
+
+    navigation.navigate(initialRoute);
+  }, [edit, fieldKey, initialRoute, navigation, selectedSportIDs, sports]);
+
+  if (single) {
+    return (
+      <Screen>
+        <Listing
+          data={sports}
+          isLoading={isFetchingInitialSports}
+          numColumns={2}
+          renderItem={({ item }) => (
+            <CardSport
+              item={item as Sport}
+              onPress={() => onSelectSingle(item)}
+              selected={selectedSportIDs.includes(item.id)}
+              style={{ flex: 0.5, margin: theme.spacing.xxs }}
+            />
+          )}
+          onScroll={calcScrollOffset}
+          onRefresh={handleRefresh}
+          isRefreshing={isRefetchingSports}
+          style={{ marginBottom: 70 }}
+        />
+      </Screen>
+    );
+  }
+
+  return (
+    <Screen>
+      <Listing
+        data={sports}
+        isLoading={isFetchingInitialSports}
+        numColumns={2}
+        renderItem={({ item }) => (
```

Comment:
A suggestion - should we extract the renderItem and have it be conditional based on the "single" variable. This way we will have only one place to make changes to the listing properties. And if we do the same for the BottomActions (have them extracted in a const that returns conditionally) we can have only one return statement.

Reply:
Fixed. Can you give it a quick test please cause we had some conflicts here?